worlds/animal_well/client.py

```diff
@@ -216,55 +185,71 @@ def on_package(self, cmd: str, args: dict):
             self.slot_data = args.get("slot_data", {})
 
         try:
-            def parseIds(data):
+            def parse_ids(data):
                 for item in data:
                     if 'type' in item and 'text' in item:
                         if item['type'] == 'player_id':
-                            item['text'] = self.player_names[int(item['text'])]
+                            item['text'] = self.player_names[int(item['text'])] if item['text'].isnumeric() else item['text']
                         elif item['type'] == 'item_id':
-                            item['text'] = item_id_to_name[int(item['text'])]
+                            item['text'] = item_id_to_name[int(item['text'])] if item['text'].isnumeric() else item['text']
                         elif item['type'] == 'location_id':
-                            item['text'] = location_id_to_name[int(item['text'])]
-
+                            item['text'] = location_id_to_name[int(item['text'])] if item['text'].isnumeric() else item['text']
                 return data
 
             if cmd == "PrintJSON":
                 msgType = args.get("type")
                 if msgType == "Chat":
-                    display_to_client(self, "".join(o['text'] for o in parseIds(args.get("data"))))
+                    self.display_text_in_client("".join(o['text'] for o in parse_ids(args.get("data"))))
                 elif msgType == "Hint":
-                    text = "".join(o['text'] for o in parseIds(args.get("data")))
-                    display_to_client(self, text)
+                    text = "".join(o['text'] for o in parse_ids(args.get("data")))
+                    self.display_text_in_client(text)
                 elif msgType == "Join":
-                    display_to_client(self, args.get('data')[0]['text'])
+                    self.display_text_in_client(args.get('data')[0]['text'])
                 elif msgType == "CommandResult":
-                    display_to_client(self, args.get('data')[0]['text'])
+                    pass
                 elif msgType == "Tutorial":
                     pass
+                elif msgType == "ItemCheat":
+                    self.display_text_in_client(args.get("data")[0]["text"])
                 elif msgType == "ItemSend":
-                    data = parseIds(args.get('data'))
-                    text = "".join(o['text'] for o in data).replace(')', '').replace('(', 'at ')
-                    display_to_client(self, text)
+                    destination_player_id = args["receiving"]
+                    source_player_id = args["item"][2]  # it's a tuple, so we can't index by name
+                    self_slot: int = self.slot
+                    # we don't want to display every item message, just ones relevant to us
+                    if self_slot not in [source_player_id, destination_player_id]:
```

Comment:
[ctx.slot_concerns_self()](https://github.com/ArchipelagoMW/Archipelago/blob/main/CommonClient.py#L461)
to catch item links as well

Reply:
Ooh [is_uninteresting_item_send()](https://github.com/ArchipelagoMW/Archipelago/blob/4620493828b73657633f4ec8d94dd91d4049c4e4/CommonClient.py#L473)

Reply:
that works too haha